manager/integration/tests/test_settings.py

```diff
@@ -653,11 +656,78 @@ def test_setting_backing_image_auto_cleanup():  # NOQA
         2. The related backing image file is removed.
         3. The download state of other disks keep unchanged.
            All volumes still work fine.
-    8. Delete all volumes. Verify that all states in the backing image disk map
-       will become terminating first,
-       and all entries will be removed from the map later.
+    8. Delete all volumes. Verify that there will only remain 1 entry in the
+       backing image disk map
     9. Delete the backing image.
     """
+    # Step 1
+    update_setting(client,
+                   "backing-image-cleanup-wait-interval",
+                   BACKING_IMAGE_CLEANUP_WAIT_INTERVAL)
+
+    # Step 2
+    create_backing_image_with_matching_url(
+            client, BACKING_IMAGE_NAME, BACKING_IMAGE_QCOW2_URL)
+
+    # Step 3
+    volume_names = [
+        volume_name + "-1",
+        volume_name + "-2",
+        volume_name + "-3"
+    ]
+
+    for volume_name in volume_names:
+        volume = create_and_check_volume(
+                            client, volume_name, 3,
+                            str(BACKING_IMAGE_EXT4_SIZE),
+                            BACKING_IMAGE_NAME)
+
+    # Step 4
+    lht_host_id = get_self_host_id()
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        volume.attach(hostId=lht_host_id)
+        wait_for_volume_healthy(client, volume_name)
+        assert volume.backingImage == BACKING_IMAGE_NAME
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    assert len(backing_image.diskFileStatusMap) == 3
+    for disk_id, status in iter(backing_image.diskFileStatusMap.items()):
+        assert status.state == "ready"
+        assert disk_id != ''
+
+    backing_images_in_disk = os.listdir("/var/lib/longhorn/backing-images")
+    assert len(backing_images_in_disk) == 1
+
+    # Step 5
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        volume.updateReplicaCount(replicaCount=1)
+
+    # Step 6
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        for replica in volume.replicas:
+            if replica.hostId == lht_host_id:
+                replica_name = replica.name
+        volume.replicaRemove(name=replica_name)
+
+    time.sleep(int(BACKING_IMAGE_CLEANUP_WAIT_INTERVAL))
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 3, "ready")
+
+    # Step 7
+    update_setting(client, "backing-image-cleanup-wait-interval", "1")
+    time.sleep(int(BACKING_IMAGE_CLEANUP_WAIT_INTERVAL))
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 2, "ready")
+
+    # Step 8
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        client.delete(volume)
+        wait_for_volume_delete(client, volume_name)
+
+    time.sleep(int(BACKING_IMAGE_CLEANUP_WAIT_INTERVAL))
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
```

Comment:
This should be `check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 0, "ready")`? because all volumes gone, and all backing images are not referenced by any replicas anymore, the info should be all cleaned up?

Reply:
cc @shuo-wu

Reply:
Yeah, I remembered that now. Thanks, @shuo-wu and @cchien816 .